tools_for_todoist/models/event.py

```diff
@@ -121,15 +126,46 @@ def _last_occurrence(self):
         if not is_allday(start):
             last_occurence = last_occurence.astimezone(start.tzinfo)
         return last_occurence.date() if is_allday(start) else last_occurence
+    
+    def _find_next_occurence(self, rrule_instances):
+        first_exception_start = min(
+            (
+                start
+                for start in (
+                    x._get_start() 
+                    for x in self.exceptions.values()
+                    if not x.is_cancelled
+                )
+                if start >= (datetime.now() if is_allday(start) else datetime.now(UTC))
+            ),
+            default=None,
+        )
+        invalid_starts = [
+            x._get_original_start()
+            for x in self.exceptions.values()
+        ]
+        first_moment = datetime.now() if is_allday(self._get_start()) else datetime.now(UTC)
+        inc = True
+        for _ in range(len(self.exceptions) + 1):
```

Comment:
I don't like how convoluted this logic is.
1) `next_regular_occurence` is not instantiated and you rely on getting inside this `for` at least one to set it
2) you are looping over the number of exceptions which sounds completely unnecessary and illogical.
3) The moment you go over `first_exception_start` while looping occurrences, you should just return `first_exception_start`

Reply:
> you are looping over the number of exceptions which sounds completely unnecessary and illogical.

My point was having an upper limit of the loop iterations, but it's unnecessary